src/generic/datavgen.cpp

```diff
@@ -4500,16 +4496,27 @@ bool wxDataViewMainWindow::TryAdvanceCurrentColumn(wxDataViewTreeNode *node, wxK
 
     if ( node )
     {
-        // navigation shouldn't work in branch nodes without other columns:
-        if ( node->HasChildren() && !GetModel()->HasContainerColumns(node->GetItem()) )
+        // navigation shouldn't work in nodes with fewer than two columns
+        numColsWithValue = 0;
+        unsigned int cols = GetOwner()->GetColumnCount();
+        for ( unsigned int i = 0; i < cols; i++ )
+            if ( GetModel()->HasValue(node->GetItem(), i) )
+                numColsWithValue++;
+        if (numColsWithValue<2)
```

Comment:
As this fragment is being used at least twice, it would be preferable to add a function, e.g. `DoesColumnHaveSingleItemOnly()` (but please feel free to use a better name if you see one) and call it from both places.

Reply:
Done. By creating a function HasItemMultipleValues I have moved the new code out of here and left the logic here virtually identical to the original one. See commit 20243534ca9853a2d13a678379fe0a46398f01cf.